receiver/vcenterreceiver/metrics.go

```diff
@@ -140,6 +141,23 @@ func (v *vcenterMetricScraper) recordVMStats(
 		return
 	}
 	v.mb.RecordVcenterVMCPUUtilizationDataPoint(ts, 100*float64(cpuUsage)/float64(cpuLimit))
+
+	// OverallCpuReadiness is only available in vSphere API 7.0
+	// https://vdc-repo.vmware.com/vmwb-repository/dcr-public/d1902b0e-d479-46bf-8ac9-cee0e31e8ec0/07ce8dbd-db48-4261-9b8f-c6d3ad8ba472/vim.vm.Summary.QuickStats.html
+	vsphereAPIVersion, err := version.NewVersion(v.client.vimDriver.ServiceContent.About.ApiVersion)
```

Comment:
is this making a request for every VM? might be overkill. Could you just check if cpuReadiness is null or something and not record it in that case?